Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp

```diff
@@ -433,3 +433,74 @@ API::ContentRuleList& webkitUserContentFilterGetContentRuleList(WebKitUserConten
     ASSERT(userContentFilter);
     return *userContentFilter->contentRuleList;
 }
+
+/**
+ * WebKitUserScriptMessage: (ref-func webkit_user_script_message_ref) (unref-func webkit_user_script_message_unref)
+ *
+ * A helper class that sends user provided script to the signal handler.
+ *
+ * Since: 2.38
+ */
+struct _WebKitUserScriptMessage {
+    _WebKitUserScriptMessage(WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler)
+        : completionHandler(completionHandler)
+        , referenceCount(1)
+    {
+        JSCContext* jscContext = jsc_context_new();
```

Comment:
It looks like you forgot to change this to use `SharedJavaScriptContext::singleton().getOrCreateContext()`, like in the other part of the patch.

Also, there is something I missed in my previous review (sorry!): we should create an undefined value here if deserializing fails, see longer explanation below.

Actually, I think we do not even need to do that, because calling a message handler without passing a parameter with `window.webkit.messageHandler.<name>.postMessage()` is the same as passing `undefined` implicitly&mdash;we need to double check this.